Change how filter works to match our Java API

pveyes opened the pull request:
See `examples/flash-filter/UseFlash` for new API signature
- Use `onRequest` and `onResponse` instead of `onFilter`
- Remove `page._requestFilters` and `page._responseFilters`, use `page.filters` public property instead
- Monkey patch `res.end` to properly intercept the response and run `onResponse` filter
- Move Sender filter to last Router arguments
- Update babel 6
- Add optional callback to `router.start` to easily know when app is ready

asendia commented on src/core/Router.js:
```diff
@@ -34,23 +35,27 @@ class Router {
     });
     routerArguments.push(page.render.bind(page));
-    page.getResponseFilters().forEach(a => {
+    page.filters.forEach(a => {
```
The filter order in our internal java library given the config:

```
@FilterA
@FilterB
```

works in this order:
1. FilterA.onRequest
2. FilterB.onRequest
3. FilterB.onResponse
4. FilterA.onResponse

That's the reason why I force the user to create 2 separate filter so the execution order becomes more clear, or did I miss something?

pveyes committed:
Change how filter works to match our Java API

See `examples/flash-filter/UseFlash` for new API signature

- Use onRequest and onResponse instead of onFilter
- Remove requestFilters and responseFilters, use page.filters public property instead
- Monkey patch res.end to properly intercept the response
- Move Sender filter to last Router arguments instead
- Update babel 6

pveyes commented:
- Monkey patch `res.end` instead so we don't need to call `next()` after `res.redirect()`
- Reverse onResponse filter

PS force pushed
